# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/WinUtils.h
# Commit: 75fb2711fae9
# Full Hash: 75fb2711fae94f6198683d3e821b82d0081ff2af
# Author: Daisuke Akatsuka <daisuke@birchill.co.jp>
# Date: 2025-04-23 09:21:27
# Regressor Bug: 1915762
# File Overlap Count: 1
# Description:
#   Bug 1915762: Apply getFaviconForPage() in WinUtils r=win-reviewers,mak,handyman
#   
#   Depends on D240120
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D240566
# ==============================================================================

diff -r 579ec95712da -r 75fb2711fae9 widget/windows/WinUtils.h
--- a/widget/windows/WinUtils.h	Wed Apr 23 04:40:34 2025 +0000
+++ b/widget/windows/WinUtils.h	Wed Apr 23 04:40:34 2025 +0000
@@ -581,34 +581,6 @@
 
 typedef MozPromise<nsString, nsresult, true> ObtainCachedIconFileAsyncPromise;
 
-#ifdef MOZ_PLACES
-class AsyncFaviconDataReady final : public nsIFaviconDataCallback {
- public:
-  NS_DECL_ISUPPORTS
-  NS_DECL_NSIFAVICONDATACALLBACK
-
-  AsyncFaviconDataReady(
-      nsIURI* aNewURI, RefPtr<nsISerialEventTarget> aIOThread,
-      const bool aURLShortcut, already_AddRefed<nsIRunnable> aRunnable,
-      UniquePtr<MozPromiseHolder<ObtainCachedIconFileAsyncPromise>>
-          aPromiseHolder = nullptr);
-  nsresult OnFaviconDataNotAvailable(void);
-
- private:
-  ~AsyncFaviconDataReady() {
-    if (mPromiseHolder) {
-      mPromiseHolder->RejectIfExists(NS_ERROR_FAILURE, __func__);
-    }
-  }
-
-  nsCOMPtr<nsIURI> mNewURI;
-  RefPtr<nsISerialEventTarget> mIOThread;
-  nsCOMPtr<nsIRunnable> mRunnable;
-  UniquePtr<MozPromiseHolder<ObtainCachedIconFileAsyncPromise>> mPromiseHolder;
-  const bool mURLShortcut;
-};
-#endif
-
 /**
  * Asynchronously tries add the list to the build
  */
